r/R/dataset-format.R

```diff
@@ -271,26 +264,74 @@ check_ambiguous_options <- function(passed_opts, opts1, opts2) {
   }
 }
 
+check_schema <- function(schema, column_names) {
```

Comment:
This function is all existing code which I just extracted out into its own function